src/components/Characters.js

```diff
@@ -0,0 +1,57 @@
+import "./Characters.css";
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+export default function Characters() {
+  const [characters, setCharacters] = useState([]);
+  const [page, setPage] = useState(1);
+  const [allPages, setAllPages] = useState();
+
+  useEffect(() => {
+    const url = `https://rickandmortyapi.com/api/character/?page=${page}`;
+
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        setCharacters((initialCharacter) => {
+          return [...initialCharacter, ...data.results];
+        });
+        setAllPages(data.info.pages);
+      });
+  }, [page]);
+
+  function handleLoadMore() {
+    if (page < allPages) {
+      setPage(page + 1);
+    }
+  }
+
+  function createCharacters() {
+    return characters.map((character) => {
+      const { id, name, image } = character;
+      return (
+        <div key={id}>
+          <Link to={`/characters/${id}`}>
+            <div class="Character-Grid">
+              <figure>
+                <img className="Character-Image" src={image} alt={name} />
+                <figcaption className="Character-Text">{name}</figcaption>
+              </figure>
```

Comment:
oh, that's new to me. Figure and figcaption. Cool. Inspiration.

Reply:
These are merely workarounds. I used them to have the text and photo together - didn't really work though. The figure-thing was an just an idea.